src/StorageSync/StorageSync/ServerEndpoint/SetServerEndpointCommand.cs

```diff
@@ -192,36 +193,55 @@ public override void ExecuteCmdlet()
                 var storageSyncServiceName = default(string);
                 var parentResourceName = default(string);
 
-                if (this.IsParameterBound(c => c.ResourceId))
-                {
-                    var resourceIdentifier = new ResourceIdentifier(ResourceId);
-                    resourceName = resourceIdentifier.ResourceName;
-                    resourceGroupName = resourceIdentifier.ResourceGroupName;
-                    parentResourceName = resourceIdentifier.GetParentResourceName(StorageSyncConstants.SyncGroupTypeName, 0);
-                    storageSyncServiceName = resourceIdentifier.GetParentResourceName(StorageSyncConstants.StorageSyncServiceTypeName, 1);
-                }
-                else if (this.IsParameterBound(c => c.InputObject))
+                var updateParameters = new ServerEndpointUpdateParameters();
+
+                if (this.IsParameterBound(c => c.InputObject))
                 {
                     resourceName = InputObject.ServerEndpointName;
                     resourceGroupName = InputObject.ResourceGroupName;
                     parentResourceName = InputObject.SyncGroupName;
                     storageSyncServiceName = InputObject.StorageSyncServiceName;
+
+                    updateParameters.CloudTiering = InputObject.CloudTiering;
+                    updateParameters.VolumeFreeSpacePercent = InputObject.VolumeFreeSpacePercent;
+                    updateParameters.TierFilesOlderThanDays = InputObject.TierFilesOlderThanDays;
```

Comment:
Shouldn't there be a test update to ensure that this property has a value in the object after New/Get?

Reply:
It would probably require re-recording the test to validate this - if this would keep this from making it in by code complete, lmk, and please file a test debt issue for StorageSync to add this test by next release.

Reply:
https://github.com/Azure/azure-powershell/issues/9686